sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/ProjectBuilder.cs

```diff
@@ -64,8 +64,20 @@ public Compilation GetCompilation(ParseOptions parseOptions = null, CompilationO
         public Document FindDocument(string name) =>
             Project.Documents.Single(d => d.Name == name);
 
-        public ProjectBuilder AddReferences(IEnumerable<MetadataReference> references) =>
-            FromProject(Project.AddMetadataReferences(references ?? Enumerable.Empty<MetadataReference>()));
+        public ProjectBuilder AddReferences(IEnumerable<MetadataReference> references)
+        {
+            var existingReferences = Project.MetadataReferences;
```

Comment:
Wouldn't be better to fix this in the test setup? This will allow us to be sloppy with metadata references.

Reply:
there's a lot of tests to fix...

Reply:
After reverting, I got NN errors like below
```
  Message: 
    Test method SonarAnalyzer.UnitTest.Common.SecurityHotspotTest.SecurityHotspotRules_AreDisabledByDefault threw exception: 
    System.ArgumentException: Specified sequence has duplicate items
    Parameter name: metadataReferences[19]
```
![image](https://user-images.githubusercontent.com/38876598/97424372-e6f63b80-1910-11eb-8f85-8b5c678dc331.png)

Reply:
I see... I think the effort is to big and it's not worth it. Maybe we will take a second look after we change the NuGet api.

Reply:
I put a comment in #3674 to keep it in mind https://github.com/SonarSource/sonar-dotnet/issues/3674#issuecomment-717857880